### PR TITLE
Vagrant setup

### PR DESCRIPTION
**Pull request by tpenning**
The Vagrantfile defines the controller and workers configuration and provides all nodes with their default suggested parameters in an easily modifiable way. Additionally, a short description on how to run this with Vagrant was added to the README.md.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# Configure the VM
+Vagrant.configure("2") do |config|
+    # The box and version used
+    config.vm.box = "bento/ubuntu-24.04"
+    config.vm.box_version = "202404.26.0"
+
+    # Define the controller node
+    config.vm.define "controller" do |controller|
+        controller.vm.network "private_network", ip: nodes_ip + "2"
```
Comment (RodinHaker) [inline Vagrantfile]: The ip part in this could be replaced with `"#{nodes_ip}2"` iirc, but it's a detail

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        controller.vm.provider "virtualbox" do |v|
+            v.cpus = controller_cores
+            v.memory = controller_mem
+        end
+    end
+
+    # Define the worker nodes
+    (1..workers).each do |i|
+        config.vm.define "node#{i}" do |worker|
+            worker.vm.network "private_network", ip: nodes_ip + "#{2 + i}"
```
Comment (RodinHaker) [inline Vagrantfile]: same here: ` "#{nodes_ip}#{2 + i}"`

**Review by RodinHaker — APPROVED** [review]
Beyond the little nitpicky comments, this looks good to me!

File: Vagrantfile (hunk 1 above)
Comment (tpenning) [reply Vagrantfile]: Improved this

File: Vagrantfile (hunk 2 above)
Comment (tpenning) [reply Vagrantfile]: Improved this

**Pull request merged**